File: app/src/main/java/com/otnam/kasirqiu/MainActivity.java
package com.otnam.kasirqiu;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.RecyclerView;

import android.app.ProgressDialog;
import android.content.ClipData;
import android.content.Intent;
import android.os.Bundle;
import android.text.TextUtils;
import android.text.method.HideReturnsTransformationMethod;
import android.text.method.PasswordTransformationMethod;
import android.view.MotionEvent;
import android.view.View;
import android.widget.Adapter;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import com.google.android.gms.auth.api.identity.BeginSignInRequest;
import com.google.android.gms.auth.api.identity.Identity;
import com.google.android.gms.auth.api.identity.SignInClient;
import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;
import com.google.android.material.floatingactionbutton.ExtendedFloatingActionButton;
import com.google.firebase.auth.AuthResult;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {
    TextView reg;
    Button buttonLogin;
    EditText email, password;
    boolean passwordVisible;
    FirebaseAuth auth;
    ProgressDialog loadingBar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        reg = findViewById(R.id.new_account);
        email=findViewById(R.id.field_email);
        password=findViewById(R.id.field_password);
        buttonLogin=findViewById(R.id.btn_login);
        auth=FirebaseAuth.getInstance();
        loadingBar= new ProgressDialog(this);

        password.setOnTouchListener(new View.OnTouchListener() {
            @Override
            public boolean onTouch(View v, MotionEvent event) {
                final int Right=2;
                if(event.getAction()==MotionEvent.ACTION_UP){
                    if(event.getRawX()>=password.getRight()-password.getCompoundDrawables()[Right].getBounds().width()){
                        int selection=password.getSelectionEnd();
                        if(passwordVisible){
                            //set drawable image here
                            password.setCompoundDrawablesRelativeWithIntrinsicBounds(0,0,R.drawable.ic_baseline_visibility_off_24, 0);
                            //for hide password
                            password.setTransformationMethod(PasswordTransformationMethod.getInstance());
                            passwordVisible=false;
                        }else{
                            //set drawable image here
                            password.setCompoundDrawablesRelativeWithIntrinsicBounds(0,0,R.drawable.ic_baseline_visibility_24, 0);
                            //for show  password
                            password.setTransformationMethod(HideReturnsTransformationMethod.getInstance());
                            passwordVisible=true;
                        }
                        password.setSelection(selection);
                        return true;
                    }
                }
                return false;
            }
        });

        buttonLogin.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String sEmail=email.getText().toString();
                String sPassword=password.getText().toString();

                loginMethode(sEmail,sPassword);
            }

            private void loginMethode(String sEmail, String sPassword) {
                if(TextUtils.isEmpty(sEmail)){
                    email.setError("Email is required");
                }
                if(TextUtils.isEmpty(sPassword)){
                    password.setError("Password is required");
                }
                else {
                    loadingBar.setTitle("Login Progress");
                    loadingBar.setMessage("Please Wait");
                    loadingBar.show();

                    auth.signInWithEmailAndPassword(sEmail, sPassword).addOnCompleteListener(new OnCompleteListener<AuthResult>() {
                        @Override
                        public void onComplete(@NonNull Task<AuthResult> task) {
                            if(task.isSuccessful()){
                                Toast.makeText(getApplicationContext(),"Login Successfully", Toast.LENGTH_SHORT).show();
                                startActivity(new Intent(getApplicationContext(), HomeScreen.class));
                                loadingBar.dismiss();
                            }
                            else {
                                Toast.makeText(getApplicationContext(),"Login Failed",Toast.LENGTH_SHORT).show();
                                loadingBar.dismiss();
                            }
                        }
                    });
                }
            }
        });

        reg.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(MainActivity.this, RegisterActivity.class));
            }
        });

    }
}

File: app/src/main/java/com/otnam/kasirqiu/MainActivity.kt
package com.otnam.kasirqiu

import android.app.ProgressDialog
import android.content.Intent
import android.os.Bundle
import android.text.TextUtils
import android.text.method.HideReturnsTransformationMethod
import android.text.method.PasswordTransformationMethod
import android.view.MotionEvent
import android.view.View
import android.view.View.OnTouchListener
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.google.firebase.auth.FirebaseAuth
import com.otnam.kasirqiu.helper.Constant
import com.otnam.kasirqiu.helper.PreferenceHelper

class MainActivity : AppCompatActivity() {
    var reg: TextView? = null
    var buttonLogin: Button? = null
    var email: EditText? = null
    var password: EditText? = null
    var passwordVisible = false
    var auth: FirebaseAuth? = null
    var loadingBar: ProgressDialog? = null

    lateinit var sharedPref: PreferenceHelper

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        supportActionBar!!.hide()
        reg = findViewById(R.id.new_account)
        email = findViewById(R.id.field_email)
        password = findViewById(R.id.field_password)
        buttonLogin = findViewById(R.id.btn_login)
        auth = FirebaseAuth.getInstance()
        loadingBar = ProgressDialog(this)
        sharedPref = PreferenceHelper(this)

        password?.let {
            it.setOnTouchListener(OnTouchListener { v, event ->
                val Right = 2
                if (event.action == MotionEvent.ACTION_UP) {
                    if (event.rawX >= it.getRight() - it.getCompoundDrawables()[Right].bounds.width()) {
                        val selection = it.getSelectionEnd()
                        passwordVisible = if (passwordVisible) {
                            //set drawable image here
                            it.setCompoundDrawablesRelativeWithIntrinsicBounds(
                                0,
                                0,
                                R.drawable.ic_baseline_visibility_off_24,
                                0
                            )
                            //for hide password
                            it.setTransformationMethod(PasswordTransformationMethod.getInstance())
                            false
                        } else {
                            //set drawable image here
                            it.setCompoundDrawablesRelativeWithIntrinsicBounds(
                                0,
                                0,
                                R.drawable.ic_baseline_visibility_24,
                                0
                            )
                            //for show  password
                            it.setTransformationMethod(HideReturnsTransformationMethod.getInstance())
                            true
                        }
                        it.setSelection(selection)
                        return@OnTouchListener true
                    }
                }
                false
            })
        }
        buttonLogin?.setOnClickListener(object : View.OnClickListener {
            override fun onClick(v: View) {
                val sEmail = email?.getText().toString()
                val sPassword = password?.getText().toString()
                loginMethode(sEmail, sPassword)
            }

            private fun loginMethode(sEmail: String, sPassword: String) {
                if (TextUtils.isEmpty(sEmail)) {
                    email?.setError("Email is required")
                } else if (TextUtils.isEmpty(sPassword)) {
                    password?.setError("Password is required")
                } else {
                    loadingBar!!.setTitle("Login Progress")
                    loadingBar!!.setMessage("Please Wait")
                    loadingBar!!.show()
                    auth!!.signInWithEmailAndPassword(sEmail, sPassword)
                        .addOnCompleteListener { task ->
                            if (task.isSuccessful) {
                                sharedPref.put(Constant.PREF_EMAIL, sEmail)
                                sharedPref.put(Constant.PREF_PASSWORD, sPassword)
                                sharedPref.put(Constant.PREF_IS_LOGIN,true)
                                Toast.makeText(
                                    this@MainActivity,
                                    "Login Successfully",
                                    Toast.LENGTH_SHORT
                                ).show()
                                startActivity(Intent(this@MainActivity, HomeScreen::class.java))
                                finish()
                                loadingBar!!.dismiss()
                            } else {
                                Toast.makeText(
                                    this@MainActivity,
                                    "Login Failed",
                                    Toast.LENGTH_SHORT
                                ).show()
                                loadingBar!!.dismiss()
                            }
                        }
                }
            }
        })
        reg?.setOnClickListener(View.OnClickListener {
            startActivity(
                Intent(
                    this@MainActivity,RegisterActivity::class.java
                )
            )
        })
    }

    override fun onStart() {
        super.onStart()
        if (sharedPref.getBoolean(Constant.PREF_IS_LOGIN)){
            startActivity(Intent(this, HomeScreen::class.java))
            finish()
        }
    }
    override fun onBackPressed() {
        super.onBackPressed();
        finishAffinity();
    }
}